src/common/config.cc

```diff
@@ -1227,6 +1227,47 @@ void md_config_t::diff(
   }
 }
 
+void md_config_t::diff_setting(const md_config_t *other,
```

Comment:
Humm, we only decided to have a different function **diff_setting()** because we would need to change the **diff()** signature and behavior if a setting was passed or not. But it is a possibility.